### PR TITLE
Port to Python 3 and PyPy, with CI.

### LINKED ISSUE #3: builds for pypy and pypy3 fail
https://travis-ci.org/github/zopefoundation/zc.set/jobs/665761717

### PR DESCRIPTION
**Pull request by jamadden**
Continuation of #2. Fixes #3.

Only the last few commits are relevant to PyPy.

### CODE REVIEW

**Comment by jamadden** [comment]
I'm seeing duplicate checks being reported. One links to Travis directly, one links to github and shows travis output. That's a new one for me.

**Comment by jamadden** [comment]
There's both the usual webhook, and a "github integration app" installed. The integration app failed mysteriously. Unlike the usual webhook, it ran on 'travis-ci.com' instead of 'travis-ci.org', which requires a new sign-in. Where'd that come from, and can we delete it? I don't think we need four sets of builds.

**Comment by mgedmin** [comment]
I prefer travis-ci.org because I hate the design of the "Checks" tab on GitHub.

**Comment by jamadden** [comment]
> I prefer travis-ci.org because I hate the design of the "Checks" tab on GitHub.

Strong agree.

**Comment by jamadden** [comment]
But I don't know why it's here. There are no configured actions. The 'integrations' tab of the settings for this repo does show Travis CI, but clicking 'configure' takes me to the *organization* settings. Without knowing more than I do, I'm loath to mess with that.

**Comment by mgedmin** [comment]
Somebody clicked a button on travis-ci.com and activated this repository.

I don't know how to deactivate the repo.  Via GitHub settings?  (In which case look under Integrations -> GitHub Apps; old-style travis-ci.org integration is done via Webhooks.)

I have disabled both "Build pushes" and "Build PRs" in https://travis-ci.com/github/zopefoundation/zc.set/settings.  That may suffice?

**Comment by mgedmin** [comment]
For context, travis-ci.org is semi-deprecated, and all users are expected to migrate to travis-ci.com.

I'm not going to until I'm forced: I see no upsides for me, and only downsides.

(Well, there are upsides: travis-ci.org sometimes hits GitHub API rate limits or something and stops updating commit statuses for a few hours.  And since it's deprecated, it's not maintained -- complain on Twitter, get a Travis CI support person asking if switching to travis-ci.com fixes your problem.)

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,10 @@
 *.egg-info/
-*.pyc
-/.eggs/
-/.tox/
+*.py[co]
 .coverage
 .coverage.*
+.tox/
+/.eggs/
+/.tox/
```
Comment (mgedmin) [inline .gitignore:7]: Minor: it's a bit weird to see two different .tox/ exclude rules.

**Review by mgedmin — APPROVED** [review]
LGTM.

Out of scope for this PR, but maybe use this opportunity to add 3.8 to .travis.yml?

File: .gitignore (hunk 1 above)
Comment (jamadden) [reply .gitignore:7]: Whoops, I assumed it was there. The classifier was, and it was in tox.ini.  Thanks, I'll do that.

**Comment by jamadden** [comment]
> I have disabled both "Build pushes" and "Build PRs" in https://travis-ci.com/github/zopefoundation/zc.set/settings. That may suffice?

Looks like it did! Thanks.

**Pull request merged**